src/libPMacc/include/algorithms/math/floatMath/floatingPoint.tpp

```diff
@@ -60,6 +72,29 @@ struct Float2int_rd<float>
     }
 };
 
+template<>
+struct Float2int_rn<float>
+{
+    typedef int result;
+
+    HDINLINE result operator( )(float value)
+    {
+#if __CUDA_ARCH__
+        return ::__float2int_rn( value );
+#else
+        if(value < 0.0f)
+            return -(*this)(-value);
+        static const result epsilon = 1e-7f;
+        /* Round towards nearest with .5 -> +inf */
+        result res = static_cast<int>(value + 0.5f);
+        /* If we were close to x.5 then make sure res is even */
+        if( abs(0.5f - (res - value)) < epsilon )
+            return res & ~1;
```

Comment:
Well it is simply the definition of this rounding mode: Round "normally" and fix bias: If we are at .5 then cancel out the "odd" bit so it is even.